docs/docs/guides/persistence.mdx

```diff
@@ -0,0 +1,58 @@
+---
+title: State Persistence
+sidebar_label: State Persistence
+---
+
+
+Recoil makes it easy to persist your state.
+
+In order to persist your state you need two simple steps:
+
+First observe the changes to your atoms using useTransactionObservation hook (note: This is in unstable state as of now),
+it will tell you about what changed and you can simply save these changes to the storage of your preference.
+This is a basic example implementation:
+
+```javascript
+useTransactionObservation_UNSTABLE(({ atomValues, atomInfo, modifiedAtoms, transactionMetadata }) => {
+        for (const modifiedAtom of modifiedAtoms) {
+            Storage.setItem(modifiedAtom, JSON.stringify({ value: atomValues.get(modifiedAtom) }));
+        }
+    })
+```
+
+*Storage* could be [*LocalStorage*](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage), *[AsyncStorage](https://github.com/react-native-community/react-native-async-storage)* or whichever storage you like.
+
+* *atomValues*: Gives you information about the current state of your atoms.
+* *modifiedAtoms* Gives you a map containing the modified atoms.
+* *transactionMetaData* Gives you metadata about the current change.
+
+
+After you ensure that you're saving your state to your storage, you need to recover it, it's easy using the prop *initializeState* on *RecoilRoot* component.
+It's a function that provides a set method to update atoms given a key.
+You just need to pass the atom key and the value you stored to this method and it will update it to the correct state.
+Here is a basic example:
+
+
+```javascript
+const initializeState = ({ set }) => {
+        Storage.getAllKeys((error, keys) => {
+            const promises = keys.map((key) =>
+                Storage.getItem(key)
+            );
+            Promise.all(promises).then((values) => {
+                for (let i = 0; i < promises.length; i++) {
+                    const key = keys[i];
+                    const value = JSON.parse(values[i]).value;
+                    set({ key }, value);
+                }
+                setIsReady(true);
```

Comment:
what `setIsReady` is doing? should there be a useState?

Reply:
Yes, it's a useState setter, should I remove it or add the useState before the method?

Reply:
I think its better to remove it,

Reply:
Done.